### ISSUE TITLE
File list refresh needs it's own thread

### ISSUE THREAD

**Issue by mowgli** [BestImageViewer/geeqie#115]
When you have a big list of files, like 5000 or more and one is updated or deleted by another program, the screen goes blank
for the entire time the list is refreshing, instead of it moving to the next pic and displaying that.
WHY IS THIS IMPORTANT?
Because if you have something putting in a pic every second and you want to watch near realtime YOU CAN'T!!

Reported by: *anonymous

**Comment by archels** [comment]
Bump. The speed of file deletion in geeqie is just tragic. Probably this is due to the same.

(Altogether, thanks for making geeqie! You rule!)

**Comment by caclark** [comment]
> The speed of file deletion in geeqie is just tragic. Probably this is due to the same.

With a folder of 6000 images, each of 20GB, I don't see this problem.
Which version of Geeqie are you using, GTK2 or GTK3, and do you thumbnails enabled? And any other settings that might influence this?

**Comment by archels** [comment]
Hi @caclark, thanks for your reponse.

I'm on geeqie 1.3 / GTK2. No thumbs. Does it make sense to try compiling master and/or using GTK3?

**Comment by caclark** [comment]
I replicated your use case but did not see a problem.

>  Does it make sense to try compiling master and/or using GTK3?

If you have time, and you are using a Debian-type system, you could:

`wget https://raw.githubusercontent.com/BestImageViewer/geeqie/master/web/geeqie-install-debian.sh`
Copy it into /tmp and run it. You get the choice of installing for either GTK2 or GTK3.

**Comment by alerque** [comment]
@caclark I agree this should have been dealt with years ago. In the mean time you might review [my solution using hooks](https://github.com/alerque/que/tree/master/.config/vcsh/hooks-enabled) that handles exactly this situation. The tricky thing with setting up a new system is that you need these hooks in place _before_ cloning anything, and the hooks themselves are in my repos. To deal with the catch 22, I use a [bootstrap script that downloads the hooks](https://github.com/alerque/que/blob/master/bin/que-home-bootstrap.bash#L50) before cloning any vcsh repos. It also takes care of getting SSH keys in place to clone other vcsh repositories, but that is out of scope for this issue report.
